File: src/1-500/007/ReverseInteger.py
class Solution:
    def reverse(self, x: int) -> int:
        sign = 1 if x >= 0 else -1
        y = x * sign
        z = str(y)
        z = z[::-1]
        z = int(z) * sign
        if (z < 2**31-1 and z > -2**31):
            return z
        else:
            return 0
        

File: src/1-500/007/ReverseInteger.java
class Solution {
    public int reverse(int x) {
		long y = 0;
		int sign = 1;
		if (x < 0) {
			sign = -1;
			x = -x;
		}
		while (x != 0) {
			y = y * 10 + x % 10;
			if (sign * y > Integer.MAX_VALUE || sign * y < Integer.MIN_VALUE) {
				return 0;
			}
			x /= 10;
		}
		return (int) (y * sign);

	}
}